### PR TITLE
Auto-delete Old Images on Event Update

### PR DESCRIPTION
**Pull request by kudrautsau-pavel**
**Description:**
Implemented automatic deletion of previous event images upon uploading a new image, streamlining media storage management.

**Changes:**

1. Enforced a check for new image uploads in event updates.
2. Automated the removal of old images when new ones are uploaded, using Django's storage system.

**Testing:**

Update an event with a new image and verify that the old image is removed from the server.

The change addresses media storage clutter without affecting existing functionality.

It is a temporary solution and be updated, as we move files storage to a GCP, to be updated when REST API changes will become the main branch.

### CODE REVIEW

File: entrypoint-mac.sh [hunk 1]
```diff
@@ -5,5 +5,6 @@ pip install -r requirements/local.txt
 docker-compose -f deployments/docker-compose.yml up -d
 python3 manage.py migrate --noinput
 echo "from django.contrib.auth.models import User; User.objects.filter(email='admin@example.com').delete();User.objects.create_superuser('admin', 'admin@example.com', 'admin')" | python manage.py shell
-python manage.py runserver
+python3 manage.py loaddata --app events category_fixture.json
```
[the remark this answers is not in the document]
Comment (kudrautsau-pavel) [reply entrypoint-mac.sh:8]: to populate the database  with defulat events categories

File: apps/events/views.py [hunk 2]
```diff
@@ -214,6 +247,245 @@ def post(self, request, event_id):
...
+    # Finding events within radius
+    # from django.contrib.gis.geos import Point
+    # from django.contrib.gis.measure import Distance
+    #
+    #
+    # lat = 52.5
+    # lng = 1.0
+    # radius = 10
+    # point = Point(lng, lat)
+    # Event.objects.filter(location__distance_lt=(point, Distance(km=radius)))
```
[the remark this answers is not in the document]
Comment (kudrautsau-pavel) [reply apps/events/views.py:489]: no need them, will delete them

File: apps/events/migrations/0013_create_event_categories.py [hunk 3]
```diff
@@ -23,4 +23,4 @@ def create_event_categories(apps, schema_editor):
             ]
         )
 
-    operations = [migrations.RunPython(create_event_categories)]
+    # operations = [migrations.RunPython(create_event_categories)]
```
Comment (sakalex) [inline apps/events/migrations/0013_create_event_categories.py:26]: Do not modify automatically generated files
If you want to make changes or do a rollback, create a new migration for tihs

File: apps/events/views.py [hunk 4]
```diff
@@ -65,11 +69,29 @@ def get_success_url(self):
...
         if form.is_valid():
             if form.instance.type == "private":
                 form.instance.is_visible = False
             else:
                 form.instance.is_visible = True
+
+            # Check if a new image for the event is uploaded
+            new_image_uploaded = "image" in request.FILES and request.FILES[
+                "image"
+            ] != form.initial.get("image")
```
Comment (sakalex) [inline apps/events/views.py:82]: One-line constructions are hard to read
Please use a simple if

You can use somethink like
```
if request.FILES.get("image", "") != form.initial.get("image"):
     new_image_uploaded = True
```

File: media/events/image/6d8445970bc932ac755fe1eb923f538d78935142.jpeg
Comment (sakalex) [inline media/events/image/6d8445970bc932ac755fe1eb923f538d78935142.jpeg:1]: It looks like this file can also be deleted

File: apps/events/views.py [hunk 5]
```diff
@@ -215,6 +237,225 @@ def post(self, request, event_id):
...
+    # Finding events within radius
+    # from django.contrib.gis.geos import Point
+    # from django.contrib.gis.measure import Distance
+    #
+    #
+    # lat = 52.5
+    # lng = 1.0
+    # radius = 10
+    # point = Point(lng, lat)
+    # Event.objects.filter(location__distance_lt=(point, Distance(km=radius)))
```
Comment (sakalex) [inline apps/events/views.py:457]: Why do we need this comment?

Comment (sakalex) [inline apps/events/views.py:457]: Looks like a ctr+V miss
Delete it all

**Review by sakalex — CHANGES_REQUESTED** [review]
Please review the changes yourself before asking for a review.

**Pull request closed without merging**